### PR TITLE
JAMES-2586 Rename Postgres Subscription

### PR DESCRIPTION
**Pull request by vttranlina**
rename JPA in prefix to Postgres

### CODE REVIEW

**Review by quantranhong1999 — APPROVED** [review]

File: mailbox/postgres/src/main/resources/META-INF/spring/mailbox-jpa.xml [hunk 1]
```diff
@@ -53,7 +53,7 @@
         <constructor-arg index="1" ref="jpa-sessionMapperFactory"/>
         <constructor-arg index="2" ref="event-bus"/>
     </bean>
-    <bean id="jpa-sessionMapperFactory" class="org.apache.james.mailbox.jpa.JPAMailboxSessionMapperFactory">
+    <bean id="jpa-sessionMapperFactory" class="org.apache.james.mailbox.postgres.PostgresMailboxSessionMapperFactory">
```
Comment (chibenwa) [inline mailbox/postgres/src/main/resources/META-INF/spring/mailbox-jpa.xml:56]: Remove this file: we will not assemble this mailbox with Spring...

**Comment by Arsnael** [comment]
Maybe actually we should just move all jpa subpath in packages to postgres already (in all postgres modules), even if some classes are still using jpa. Would maybe help avoid mistakes on forgetting it later?

File: mailbox/postgres/src/main/resources/META-INF/spring/mailbox-jpa.xml (hunk 1 above)
Comment (vttranlina) [reply mailbox/postgres/src/main/resources/META-INF/spring/mailbox-jpa.xml:56]: we will do it after all dependencies move to postgres, now keep it for ci pass.

**Comment by vttranlina** [comment]
When I rename package from jpa -> postgres, ci fail because can't not initial some jpa spring bean. 
I tried debugging but was not sure could fix it quickly, for ease, I dropped some "un-used" implement of `AbstractJPAMailboxMessage` 
```
JPAEncryptedMailboxMessage.java
JPAMailboxMessageWithAttachmentStorage.java
JPAStreamingMailboxMessage.java
```
(that we discussed will not implemented)

**Review by Arsnael — APPROVED** [review]

**Comment by Arsnael** [comment]
https://ci-builds.apache.org/job/james/job/ApacheJames/job/PR-1787/3/testReport/junit/org.apache.james.backends.postgres/ConnectionThreadSafetyTest/connectionShouldWorkWellWhenItIsUsedByMultipleThreadsAndQueriesIncludeBothSelectAndInsert/

Related or not?

**Comment by vttranlina** [comment]
> https://ci-builds.apache.org/job/james/job/ApacheJames/job/PR-1787/3/testReport/junit/org.apache.james.backends.postgres/ConnectionThreadSafetyTest/connectionShouldWorkWellWhenItIsUsedByMultipleThreadsAndQueriesIncludeBothSelectAndInsert/
> 
> Related or not?

not related

**Comment by Arsnael** [comment]
Restarted the build, let's see

**Review by chibenwa — APPROVED** [review]

**Pull request merged**